transaction/core/src/tx_summary.rs

```diff
@@ -0,0 +1,187 @@
+//!
+//! It's reasonable to ask, what if the computer lies to the device in the
+//! TxSummary. What if, the TxPrefix actually says one thing, and the TxSummary
+//! it gives to the device says another. The device cannot detect this if it
+//! doesn't have insight into the extended message digest. However, in this
+//! case, the transaction will simply be invalid when it is submitted, because
+//! the consensus enclave will derive a different value for the TxSummary when
+//! it attempts to validate the transaction, and that value will actually be
+//! based on the TxPrefix. So the computer can gain no advantage by lying to the
+//! device in this way.
+//!
+//! In this module we only attempt to define the TxSummary and how it is
+//! digested. In the future we may define TxSummaryUnblindingData as well, per
+//! MCIP 52.
+
+use crate::{
+    tx::{TxOut, TxPrefix},
+    CompressedCommitment, InputRules, MaskedAmount,
+};
+use alloc::vec::Vec;
+use mc_crypto_digestible::Digestible;
+use mc_crypto_keys::CompressedRistrettoPublic;
+use prost::Message;
+use serde::{Deserialize, Serialize};
+use zeroize::Zeroize;
+
+/// A subset of the data in a Tx which enables efficient verification (by e.g. a
+/// HW wallet) of the inputs and outputs of a transaction being signed.
+#[derive(Clone, Deserialize, Digestible, Eq, Hash, Message, PartialEq, Serialize, Zeroize)]
+pub struct TxSummary {
+    /// The outputs which will be added to the blockchain as a result of this
+    /// transaction
+    #[prost(message, repeated, tag = "1")]
+    pub outputs: Vec<TxOutSummary>,
+
+    /// Data in the summary associated to each real input
+    #[prost(message, repeated, tag = "2")]
+    pub inputs: Vec<TxInSummary>,
+
+    /// Fee paid to the foundation for this transaction
+    #[prost(uint64, tag = "3")]
+    pub fee: u64,
+
+    /// Token id for the fee output of this transaction
+    #[prost(uint64, tag = "4")]
+    pub fee_token_id: u64,
+
+    /// The block index at which this transaction is no longer valid.
+    #[prost(uint64, tag = "5")]
+    pub tombstone_block: u64,
+}
+
+impl TxSummary {
+    /// Make a TxSummary for a given TxPrefix and input summaries
+    pub fn new(tx_prefix: &TxPrefix, inputs: Vec<TxInSummary>) -> Self {
+        Self {
+            outputs: tx_prefix.outputs.iter().map(Into::into).collect(),
+            inputs,
+            fee: tx_prefix.fee,
+            fee_token_id: tx_prefix.fee_token_id,
+            tombstone_block: tx_prefix.tombstone_block,
+        }
+    }
+}
+
+/// A subset of the data of a TxOut.
+///
+/// Fog hint and memo are omitted to reduce size and complexity on HW device,
+/// which can't really do much with those and isn't very interested in them
+/// anyways.
+#[derive(Clone, Deserialize, Digestible, Eq, Hash, Message, PartialEq, Serialize, Zeroize)]
+pub struct TxOutSummary {
+    /// The amount being sent.
+    #[prost(oneof = "MaskedAmount", tags = "1, 6")]
+    #[digestible(name = "amount")]
```

Comment:
What does this gain us? Why not `masked_amount`?

Reply:
👍